packages/p2p/src/stores/buy-sell-store.js

```diff
@@ -254,8 +254,17 @@ export default class BuySellStore extends BaseStore {
         const { error, p2p_order_create, p2p_order_info, subscription } = order || {};
 
         if (error) {
-            setErrorMessage(error.message);
-            this.setFormErrorCode(error.code);
+            const { code, message } = error;
+
+            if (code === api_error_codes.ORDER_CREATE_FAIL_RATE_SLIPPAGE) {
+                general_store.showModal({
+                    key: 'MarketRateChangeErrorModal',
+                    props: { message },
+                });
+            } else {
+                setErrorMessage(message);
+                this.setFormErrorCode(code);
+            }
```

Comment:
for now, we check only for `OrderCreateFailRateSlippage` error code specically to show the MarketRateChangeErrorModal`

Reply:
and we still need to set the errorMessage and form error code for other scenarios such as low balance, advertiser changed rate, etc